SU2_CFD/src/output/COutput.cpp

```diff
@@ -347,15 +356,57 @@ void COutput::WriteToFile(CConfig *config, CGeometry *geometry, unsigned short f
 
     case RESTART_BINARY:
 
-      if (fileName.empty())
+      if (fileName.empty()){
         fileName = config->GetFilename(restartFilename, "", curTimeIter);
 
+        if (!config->GetWrt_Restart_Overwrite()){
+          if (config->GetMultizone_Problem()){
+            /* --- do nothing --- */  
+          }
+          else {
+
+            // append previous iteration to filename
+            filename_iter = fileName;
+            // for the final iteration we need to add 1
+            std::stringstream inner_iter_ss;
+            if (config->GetnIter() == curInnerIter+1){ 
+              inner_iter_ss << "_" << std::setw(8) << std::setfill('0') << previous_inner_iteration+1;
+              filename_iter.append((inner_iter_ss).str());
+            }
+            else {   
+              inner_iter_ss << "_" << std::setw(8) << std::setfill('0') << previous_inner_iteration;
+              filename_iter.append(inner_iter_ss.str());
+            }
+            filename_iter.append(CSU2BinaryFileWriter::fileExt);
+
+             /*--- rename previous restart.dat to restart_xxxx.dat ---*/
+            rename(restartFilename.c_str(),filename_iter.c_str()); 
+          }
+        }
+      }
+      
       if (rank == MASTER_NODE) {
-          (*fileWritingTable) << "SU2 restart" << fileName + CSU2BinaryFileWriter::fileExt;
+        if (!config->GetWrt_Restart_Overwrite())
+          (*fileWritingTable) << "moving previous restart to:" << filename_iter ;
+
+        (*fileWritingTable) << "SU2 restart" << fileName + CSU2BinaryFileWriter::fileExt;
       }
 
       fileWriter = new CSU2BinaryFileWriter(fileName, volumeDataSorter);
 
+      if ((!config->GetWrt_Restart_Overwrite()) && (config->GetnIter() == curInnerIter+1)) {
+        std::stringstream inner_iter_ss;
+        inner_iter_ss << "_" << std::setw(8) << std::setfill('0') << curInnerIter;
+        filename_iter = fileName;
+        filename_iter.append(inner_iter_ss.str());
+        fileWriterFinalIteration = new CSU2BinaryFileWriter(filename_iter, volumeDataSorter);
+
```

Comment:
I don't get why this is needed.

Reply:
Because I want the file restart_0099.dat and else we will just have restart.dat at iteration 99. It will mainly be important for the paraview files to get a nice sequence, but I would rather have SU2 write an additional file than do the copy/rename myself all the time. 

We could also simply write a restart.dat, and also write a restart_xxxx.dat every WRT_FREQ iterations, so we write it twice. There is additional writing, but it will make the implementation a bit cleaner. This is how certain commercial flow solvers do it as well. 
On the one hand I like that we can skip a complete file write, on the other I do not like that we are WRT_FREQ behind in writing the sequence.

Reply:
Ok but this logic fails if the solver converges before reaching the maximum number of iterations (so why bother? if someone wants to make a steady state movie they can rename one file, or symlink, etc.).
It looks too clunky having to create another file writer just to write the same thing with a different name, maybe add a method to the base file writer to modify the name and then call it again.

We do not have enough people maintaining the code, hence my insistence in making this as simple as possible.

Reply:
Easiest is to just write (and not rename) 2 files all the time: the restart.dat (or other files) and the same file with the iteration number appended. Writing is slower than renaming, but it will make the code simpler.